Entities/HDFirearm.cpp

```diff
@@ -850,16 +856,18 @@ void HDFirearm::Update()
 
                     // ##@#@@$ TEMP
                     shellVel.SetXY(pRound->GetShellVel(), 0);
-                    shellVel.DegRotate(degAimAngle + 150 * (m_HFlipped ? -1 : 1) + shellSpread);
+                    shellVel.DegRotate(degAimAngle + m_ShellEjectAngle * (m_HFlipped ? -1 : 1) + shellSpread);
                     pShell->SetVel(m_Vel + shellVel);
                     pShell->SetRotAngle(m_Rotation.GetRadAngle());
                     pShell->SetAngularVel(pShell->GetAngularVel() + (m_ShellAngVelRange * RandomNormalNum()));
 					pShell->SetHFlipped(m_HFlipped);
-//                  // Set the ejected shell to not hit this HeldDevice's parent, if applicable
-//                  if (m_FireIgnoresThis)
-//                      pParticle->SetWhichMOToNotHit(pRootParent, 1.0f);
+					// Set the ejected shell to not hit this HeldDevice's parent, if applicable
+					if (m_FireIgnoresThis)
+						pShell->SetWhichMOToNotHit(this, 1.0f);
                     // Set the team so alarm events that happen if these gib won't freak out the guy firing
                     pShell->SetTeam(m_Team);
+					// Set this to ignore team hits in case it's lethal
+					pShell->SetIgnoresTeamHits(true);
```

Comment:
What if a modder wanted the bullets to not ignore team hits? That's hardcoded as well, and damn right more likely to be a modder's desire than exploding shells! (no offense lol) I was just following ~~orders~~ examples.

Anyway, this'll be a key factor in how I've nerfed diggers on the data branch. Call it hacky, but I explained myself to Maxim before, so just ask him to copy+paste my answers to you if you have questions about it :P

Reply:
Hmmm, fair point haha. I still think what I said above - i.e. put a smart default rather than completely ignoring ini values - would be the better solution though. The same concept applies to bullets tbh, doing it this way would mean no more need for a lua script to set one value if you just want some gun to do friendly fire.

Anyway, feel free to leave it as-is for now if you don't wanna deal with that though. Just chuck in an issue or a TODO about it so it's not forgotten.